app/src/main/java/com/sdpteam/connectout/mapList/MapListModel.java

```diff
@@ -0,0 +1,55 @@
+package com.sdpteam.connectout.mapList;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+import androidx.lifecycle.MutableLiveData;
+
+import com.google.firebase.database.DataSnapshot;
+import com.google.firebase.database.DatabaseError;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.firebase.database.Query;
+import com.google.firebase.database.ValueEventListener;
+import com.sdpteam.connectout.event.Event;
+import com.sdpteam.connectout.event.EventCreatorModel;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MapListModel implements MapListModelManager {
+    private final DatabaseReference database;
+    private final static int MAX_EVENTS_FETCHED = 50;
+
+    private final MutableLiveData<List<Event>> cachedEvents;
```

Comment:
Nice temporary workaround for the issue we had in Firebase. But you should change the Interface `MapListModelManager` (which should be renamed as said before to `EventRepository`) to return Futures and not MutableLiveData because we should divide the ui part (LiveData) VS the model part (Futures)
See how it is done here :
https://github.com/ConnectOut-sdp/sdp2023/blob/99cd2b436a352396e367417d209a90ebf3b4b84a/app/src/main/java/com/sdpteam/connectout/profile/ProfileFirebaseDataSource.java#L26

Note that you should move those `Event` related files (our model) out of the mapList folder and have everything related to Event in the event folder ;-)